### ISSUE TITLE
YOLO-NAS quantization in a Vulkan backend

### ISSUE THREAD

**Issue by lsauerr** [ultralytics/ultralytics#22247]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I've been trying to deploy a YOLO-NAS model with a Vulkan backend to execute on a GPU.

For doing that, I have a working pipeline that performs the following:
Loads PyTorch model → Export it to ONNX → Use IREE to transform to MLIR and then compile with a Vulkan backend. 

The pipeline works and the model executes on the GPU providing numerically-accurate results for the 3 different stages. However, each inference on the embedded GPU takes too long to execute. 

In order to reduce inference time, I am trying to implement Post-Training Quantization on the system. I am not interested in doing any Quantization Aware Training for now, and I am using a sub data set to calibrate after post-training quantization.

By investigating the MLIR file generated with iree-import-onnx, I saw that most of the operands are still under float 32 bits - indicating that they may be left out of quantization for some reason. 
With that said, I have a couple of questions:
1. Do you provide any YOLO-NAS model already quantized somewhere?
2. Do you have any way to deploy YOLO models using a Vulkan backend that is a preferred method?
3. Is there any way to use something similar to model.predict() for testing the quantized model? I tried following [this ](https://colab.research.google.com/github/Deci-AI/super-gradients/blob/master/notebooks/yolo_nas_custom_dataset_fine_tuning_with_qat.ipynb?utm_source=chatgpt.com#scrollTo=YTioUmGjB45V) notebook but it looks like it is not up-to-date. 

Many thanks for the support.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @lsauerr, thanks for opening this issue and for the detailed context 🚀 This is an automated response to help get things moving quickly—an Ultralytics engineer will also assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us better understand your Vulkan/IREE and quantization setup, please share:
- Exact model source and type (Ultralytics YOLO like YOLO11/YOLO10/etc. vs third-party YOLO-NAS)
- Ultralytics version and how you exported (command/args, opset, dynamic/static shapes)
- A small code snippet showing your export and inference steps
- IREE compile flags, target, and any quantization/calibration settings you used
- A short log excerpt and one example input/output pair
- Performance numbers before/after quantization, device details, and OS info

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a Python>=3.8 environment with PyTorch>=1.8 to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thanks again for the detailed description—once you share the specifics above, we can better advise on export/quantization expectations, reproducibility, and next steps for Vulkan-based deployment ⚙️🧪

**Comment by glenn-jocher** [comment]
Thanks for the detailed write-up—quick answers: we don’t ship pre-quantized YOLO‑NAS weights; only FP32 `.pt` models are provided and supported for inference/val/export as noted in the [YOLO‑NAS docs](https://docs.ultralytics.com/models/yolo-nas/). We don’t have a first‑party Vulkan/IREE backend; our supported high‑performance deploy paths are TensorRT, OpenVINO, TFLite, ONNX Runtime, and Sony IMX500—if INT8 is your goal and hardware allows, YOLO11 + TensorRT INT8 is the most direct route (see [TensorRT export with INT8](https://docs.ultralytics.com/integrations/tensorrt/)). To sanity‑check a quantized artifact, you can run ONNX (Q/DQ) directly via the Ultralytics runner; Vulkan-compiled binaries must be tested in your own runtime:

```bash
yolo predict model=path/to/model_quant.onnx source=path/to/images imgsz=640
```

For background on trade‑offs and workflows, our [Model Quantization overview](https://www.ultralytics.com/glossary/model-quantization) may help.
